linkage-checker-exclusion-rules.xml

```diff
@@ -23,5 +31,25 @@
     <Source><Package name="net.snowflake.client.jdbc.internal"/></Source>
     <Reason>Unknown: Likely the referenced class is not used in the jdbc package</Reason>
   </LinkageError>
+  <LinkageError>
+    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.crypto.impl.LegacyAESGCM"/></Source>
+    <Reason>Not Used, appears to be Legacy</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.jwk"/></Source>
+    <Reason>Not Used</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.crypto.bc.BouncyCastleProviderSingleton"/></Source>
+    <Reason>Not Used</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.util.X509CertChainUtils"/></Source>
+    <Reason>Not Used</Reason>
+  </LinkageError>
+  <LinkageError>
+    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.crypto.bc.BouncyCastleFIPSProviderSingleton"/></Source>
+    <Reason>Not Used</Reason>
+  </LinkageError>
```

Comment:
L35,43,47,51 can more specifically be: `<Class name=".." />` assuming those are classes and we want to be specific. Specific is usually better but this could also be a more generic:
```
  <LinkageError>
    <Source><Package name="net.snowflake.ingest.internal.com.nimbusds.jose.crypto.bc"/></Source>
    <Reason>Not Used</Reason>
  </LinkageError>
```
to combine the two BC exclusions.